### ISSUE TITLE
Company breaks Python REPL

### ISSUE THREAD

**Issue by cpitclaudel** [company-mode/company-mode#603]
Hi @dgutov,

Company seems to break continuation lines in the Python REPL.  Steps to reproduce:
- Load the following init file:
  `emacs-lisp
  (setq-default python-shell-interpreter "python3")
  (global-company-mode)
  (setq-default company-idle-delay 0.01)
  `
- `M-x run-python`
- In Python REPL, type `if True:` <kbd>RET</kbd> <kbd>SPC SPC SPC SPC</kbd> `print(1)` <kbd>RET</kbd>.
- Python replies "Indentation error"

If I don't load company-mode, the error doesn't happen.

![screenshot from 2016-10-29 10-01-10](https://cloud.githubusercontent.com/assets/8181630/19830086/a62bcd26-9dbe-11e6-8103-a82adc78e0cf.png)

**Comment by cpitclaudel** [comment]
This is related to the fact that the native completion setup fails with Python 3, but should that really cause Company to break Python-mode? The error seems fixed by the patch I suggested at Emacs bug #24401, but there might be an underlying problem.

**Comment by dgutov** [comment]
What else would you expect to happen? Company uses the built-in completion, that one sends some lines to the inferior process, and they, IIUC, result in this error.

**Comment by cpitclaudel** [comment]
Hmm, I'm not fully sure what you mean.  Do you mean that there's a way to invoke completion outside of company and reproduce this issue? If so, sorry :/

**Comment by dgutov** [comment]
If you press `C-M-i` at some strategic moments while typing, you should be able to reproduce it. Not sure which exact moment, though.

**Comment by cpitclaudel** [comment]
Closing this: I pushed a fix for Emacs' bug #24401 to emacs-25.

**Comment by evgeniysharapov** [comment]
Was this resolved ? I have exactly the same problem. Solutions were either disable `company-mode` in inferior python. Playing with `company-frontends` didn't help. Even if I make it empty list, I still get an error like this

```
>>> for x in range(10):
...     print x
  File "<stdin>", line 1
    print x
    ^
IndentationError: unexpected indent
```

**Comment by dgutov** [comment]
@evgeniysharapov Emacs version?

**Comment by evgeniysharapov** [comment]
GNU Emacs 25.2.1 (x86_64-w64-mingw32) of 2017-04-24

**Comment by dgutov** [comment]
That's odd. The original scenario works fine for me in Emacs 25.2.1.

Python 3.5.2+ (default, Sep 22 2016, 12:18:14).

Is there anything else you can add to clarify the problem?

And isn't `print x` supposed to be invalid in Python 3?

**Comment by evgeniysharapov** [comment]
Hmm. I am using Python 2.7.13
I start fresh Emacs 25.2.1 with [following `~/.emacs.d/init.el](https://gist.github.com/evgeniysharapov/e0446081501108d4e5fb384042d979c1)

The result is captured [on this video](https://webmshare.com/YJ9QV)

**Comment by dgutov** [comment]
Thanks. Could you try reproducing some of this by pressing `C-M-i` at some positions (where company-mode probably initiates completion), with `company-mode` disabled?

If you manage to do that, the next step will be `M-x report-emacs-bug`.

**Comment by dgutov** [comment]
But there's also a bunch of warnings shown in this video (check out the `*Warnings*` buffer), and yet the Python REPL buffer says "native completion setup loaded".

**Comment by evgeniysharapov** [comment]
The `complete-symbol` with `company-mode` disabled works. So, what should be in the bug ? `company-mode` breaks repl ?  Once i disable `company-mode` multiline statements work.

**Comment by dgutov** [comment]
If you `complete-symbol` after every character typed, it still works?

**Comment by evgeniysharapov** [comment]
Well, interesting, if I do it after every character, then I get the same `IndentationError`

**Comment by evgeniysharapov** [comment]
Interestingly enough it only breaks if one calls `C-M-i` on the second line, otherwise it works

**Comment by dgutov** [comment]
> Interestingly enough it only breaks if one calls C-M-i on the second line, otherwise it works

Curious, but I expected something like this.

You can `M-x report-emacs-bug` now! ;)

**Comment by dgutov** [comment]
...with a scenario how to reproduce the bug without company-mode, of course.

**Comment by dgutov** [comment]
Thanks! For posterity: https://debbugs.gnu.org/cgi/bugreport.cgi?bug=28051

**Comment by dgutov** [comment]
Should be fixed in Emacs 26.1.
